src/vs/workbench/browser/actions/layoutActions.ts

```diff
@@ -463,7 +463,7 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
 	weight: KeybindingWeight.EditorContrib - 1000,
 	handler(accessor: ServicesAccessor) {
 		const layoutService = accessor.get(IWorkbenchLayoutService);
-		layoutService.toggleZenMode();
+		layoutService.toggleZenModeOff();
```

Comment:
I'd rather we query the value of the `InEditorZenModeContext` context key dynamically here and only dispatch to `toggleZenMode` when it's true, adding a whole separate method to the core layout service to half-way duplicate an existing method isn't ideal. Alternatively we could adjust `toggleZenMode` to take the desired end state as an optional parameter, similar to how `DOMNode#classList.toggle` operates.

Reply:
You're right, and I learned some architecture today. Thanks!

I guess the issue is `InEditorZenModeContext` is stored in the layout service's private state, and its public interface doesn't support parameters to `toggleZenMode` as some args change internal state. New set of changes add `isInZenMode` to expose that and use that instead, but let me know if there's a better solution!

Reply:
> I guess the issue is InEditorZenModeContext is stored in the layout service's private state

In fact all context keys are almost always global*, but they need to be accessed via an `IContextKeyService`. You can see an example of how this looks here: https://github.com/microsoft/vscode/blob/2cf8ae8a1394e7428c840000a48c621f986a4660/src/vs/workbench/contrib/debug/browser/debugCommands.ts#L297-L302

*some context keys are "scoped", and only apply to children of specific HTML elements. This isn't the case here.

You can see the full list of context keys applied to a given HTML element via the `Developer: inspect Context Keys` command. (result is logged to the dev tools)